### PR TITLE
update of yq command to v4

### PR DESCRIPTION
**Pull request by Patryk-Stefanski**
# Description

[MGDAPI-1147 ](https://issues.redhat.com/browse/MGDAPI-1147l)  Using v4 of yq caused errors when running make cluster/prepare/delorean, this was due to a change of syntax between v3 and v4  of yq and required updating yq commands to comply with new syntax and format.

### Changes Made 
- Yq updated from version 3 to version 4.9.8(most recent at the time) because v3 of yq is end of life in August.
- Yq commands updated to comply with new syntax and format as v4 introduced new syntax which isn't compatible with v3.
- Yq and jq were added as prerequisites in the readme.

### Verification 
Run the  following make commands localy to ensure they work and match the expected results: 
- `INSTALLATION=managed-api SEMVER=1.10.1 make release/prepare`
   Expected Result :`Package manifests generated successfully in packagemanifests/managed-api-service`

- `TYPE_OF_MANIFEST=master make manifest/prodsec`
   Expected Result: `Manifest generated successfully, deleting temporary files`

- `make cluster/prepare/delorean/pullsecret`
   Expected Result : `make: Nothing to be done for 'cluster/prepare/delorean/pullsecret `

- `make test-containers/check`
   Expected Result: Images should build correctly, but there is a possibility of getting an error related to access issues.

Run the following docker command to ensure the new yq image is installed : 
 - `docker build -f openshift-ci/Dockerfile.tools`
    Expected Result : Image builds sucesfully.

## Type of change

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [ ] This change requires a documentation update 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added a test case that will be used to verify my changes 
- [ ] Verified independently on a cluster by reviewer

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -423,7 +423,8 @@ deploy/integreatly-rhmi-cr.yml:
 	sed "s/OPERATORS_IN_PRODUCT_NAMESPACE/$(OPERATORS_IN_PRODUCT_NAMESPACE)/g" | \
 	sed "s/USE_CLUSTER_STORAGE/$(USE_CLUSTER_STORAGE)/g" > config/samples/integreatly-rhmi-cr.yml
 	# Workaround until in_prow annotation can be removed from prow
-	yq w -i config/samples/integreatly-rhmi-cr.yml metadata.annotations.in_prow "IN_PROW"
+	yq e -i '.metyqadata.annotations.in_prow="IN_PROW"' config/samples/integreatly-rhmi-cr.yml
```
Comment (austincunningham) [inline Makefile:426]: looks like a typo here
```suggestion
	yq e -i '.metaadata.annotations.in_prow="IN_PROW"' config/samples/integreatly-rhmi-cr.yml
```

**Review by austincunningham — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (austincunningham) [reply Makefile:426]: typo my own suggestion :facepalm:

Comment (Patryk-Stefanski) [reply Makefile]: Typo is fixed.

**Comment by makslion** [comment]
Verified that mentioned in this PR commands are succeeding while run with yq v4.9.8 installed. Also, randomly picked several changed commands and they were valid. Looks good. You can treat this as `LGTM`, but I want to have one extra sanity check

**Comment by Boomatang** [comment]
adding the lgtm on behalf of @makslion 

/lgtm

**Comment by makslion** [comment]
@austincunningham thanks

**Pull request merged**